trajopt_examples/src/tiago_industrial.cpp

```diff
@@ -0,0 +1,264 @@
+const std::string ROBOT_SEMANTIC_PARAM = "robot_description_semantic"; /**< Default ROS parameter for robot
+                                                                          description */
+const std::string TRAJOPT_DESCRIPTION_PARAM =
+    "trajopt_description"; /**< Default ROS parameter for trajopt description */
+
+static bool plotting_ = false;
+static bool write_to_file_ = false;
+static int steps_ = 5;
+static std::string method_ = "json";
+static urdf::ModelInterfaceSharedPtr urdf_model_; /**< URDF Model */
+static srdf::ModelSharedPtr srdf_model_;          /**< SRDF Model */
+static tesseract_ros::KDLEnvPtr env_;             /**< Trajopt Basic Environment */
+
+TrajOptProbPtr jsonMethod()
+{
+  ros::NodeHandle nh;
+  std::string trajopt_config;
+
+  nh.getParam(TRAJOPT_DESCRIPTION_PARAM, trajopt_config);
+
+  Json::Value root;
+  Json::Reader reader;
+  bool parse_success = reader.parse(trajopt_config.c_str(), root);
+  if (!parse_success)
+  {
+    ROS_FATAL("Failed to load trajopt json file from ros parameter");
+  }
+
+  return ConstructProblem(root, env_);
+}
+
+int main(int argc, char** argv)
+{
+  ros::init(argc, argv, "glass_up_right_plan");
+  ros::NodeHandle pnh("~");
+  ros::NodeHandle nh;
+
+  // Initial setup
+  std::string urdf_xml_string, srdf_xml_string;
+  nh.getParam(ROBOT_DESCRIPTION_PARAM, urdf_xml_string);
+  nh.getParam(ROBOT_SEMANTIC_PARAM, srdf_xml_string);
+  urdf_model_ = urdf::parseURDF(urdf_xml_string);
+
+  srdf_model_ = srdf::ModelSharedPtr(new srdf::Model);
+  srdf_model_->initString(*urdf_model_, srdf_xml_string);
+  env_ = tesseract_ros::KDLEnvPtr(new tesseract_ros::KDLEnv);
+  assert(urdf_model_ != nullptr);
+  assert(env_ != nullptr);
+
+  bool success = env_->init(urdf_model_, srdf_model_);
+  assert(success);
+
+  std::vector<std::array<double, 3> > dims = {
+    {0.4, 1.4, 0.02},
+    {0.3, 0.02, 0.45},
+    {0.3, 0.02, 0.45},
+    {0.3, 0.52, 0.02},
+    {1.1, 0.3, 0.02},
+    {0.02, 0.2, 0.3},
+    {0.02, 0.2, 0.3},
+    {0.02, 0.2, 0.3},
+    {0.42, 0.3, 0.02},
+    {0.02, 0.1, 0.06},
+    {0.02, 0.1, 0.06},
+    {0.02, 0.1, 0.06},
+    {0.42, 0.02, 0.06},
+    {0.42, 0.02, 0.06},
+    {0.42, 0.02, 0.34},
+    {0.42, 0.02, 0.02}
+  };
+  std::vector<std::array<double, 3> > positions = {
+    {0.75, -0.1, 0.7},
+    {0.8, 0.25, 0.925},
+    {0.8, -0.25, 0.925},
+    {0.8, 0.0, 1.16},
+    {0.4, -0.65, 0.7},
+    {0.1, -0.7, 0.85},
+    {0.3, -0.7, 0.85},
+    {0.5, -0.7, 0.85},
+    {0.3, -0.65, 0.71},
+    {0.1, -0.55, 0.73},
+    {0.3, -0.55, 0.73},
+    {0.5, -0.55, 0.73},
+    {0.3, -0.5, 0.73},
+    {0.3, -0.59, 0.93},
+    {0.3, -0.79, 0.87},
+    {0.3, -0.7, 0.9}
+  };
+
+
+  // Create Plotting tool
+  tesseract_ros::ROSBasicPlottingPtr plotter(new tesseract_ros::ROSBasicPlotting(env_));
+
+  // Add sphere
+  AttachableObjectPtr obj(new AttachableObject());
+
+  auto dim_iter = dims.begin();
+  auto pos_iter = positions.begin();
+  size_t idx = 0;
```

Comment:
This is not used elsewhere, now just feels like the `for` loop lost a teeth :D 
Can we please merge this into the loop below?

Reply:
The idx is actually not required. Since we iterate using the iterators.